genesyscloud/task_management_worktype/resource_genesyscloud_task_management_worktype.go

```diff
@@ -0,0 +1,302 @@
+func createTaskManagementWorktype(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTaskManagementWorktypeProxy(sdkConfig)
+
+	taskManagementWorktype := getWorktypecreateFromResourceData(d)
+
+	// Create the base worktype
+	log.Printf("Creating task management worktype %s", *taskManagementWorktype.Name)
+	worktype, err := proxy.createTaskManagementWorktype(ctx, &taskManagementWorktype)
+	if err != nil {
+		return diag.Errorf("failed to create task management worktype: %s", err)
+	}
+
+	log.Printf("Created the base task management worktype %s", *worktype.Id)
+	d.SetId(*worktype.Id)
+
+	// Create and update (for referencing other status) the worktype statuses
+	log.Printf("Creating the task management worktype statuses of %s", *worktype.Id)
+	statuses := d.Get("statuses").(*schema.Set).List()
+	if _, err := createWorktypeStatuses(ctx, proxy, *worktype.Id, statuses); err != nil {
+		return diag.Errorf("failed to create task management worktype statuses: %v", err)
+	}
+	log.Printf("Updating the destination statuses of the statuses of worktype %s", *worktype.Id)
+	if _, err := updateWorktypeStatuses(ctx, proxy, *worktype.Id, statuses, true); err != nil {
+		return diag.Errorf("failed to update task management worktype statuses: %v", err)
+	}
+
+	// Update the worktype if 'default_status_name' is set
+	if d.HasChange("default_status_name") {
+		time.Sleep(5 * time.Second)
+		err := updateDefaultStatusName(ctx, proxy, d, *worktype.Id)
+		if err != nil {
+			return diag.Errorf("failed to update default status name of worktype: %v", err)
+		}
+	}
+
+	log.Printf("Created the task management worktype statuses of %s", *worktype.Id)
+	return readTaskManagementWorktype(ctx, d, meta)
+}
+
+// readTaskManagementWorktype is used by the task_management_worktype resource to read a task management worktype from genesys cloud
+func readTaskManagementWorktype(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getTaskManagementWorktypeProxy(sdkConfig)
+
+	log.Printf("Reading task management worktype %s", d.Id())
+
+	return gcloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		worktype, respCode, getErr := proxy.getTaskManagementWorktypeById(ctx, d.Id())
+		if getErr != nil {
+			if gcloud.IsStatus404ByInt(respCode) {
+				return retry.RetryableError(fmt.Errorf("failed to read task management worktype %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("failed to read task management worktype %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceTaskManagementWorktype())
+
+		resourcedata.SetNillableValue(d, "name", worktype.Name)
+		resourcedata.SetNillableValue(d, "description", worktype.Description)
+
+		if worktype.Division != nil {
```

Comment:
I think we have a function for readinding divisions

Reply:
of course! thanks for the catch. fixed.